drivers/pinctrl/aspeed/pinctrl-aspeed.c

```diff
@@ -475,9 +475,15 @@ int aspeed_g7_gpio_request_enable(struct pinctrl_dev *pctldev,
 	const struct aspeed_g7_pincfg *pin_cfg = pinctrl->pinmux.configs_g7;
 	const struct aspeed_g7_funcfg *funcfg = pin_cfg[offset].funcfg;
 
-	for (i = 0; i < pin_cfg[offset].nfuncfg; i++)
+	for (i = 0; i < pin_cfg[offset].nfuncfg; i++) {
+		if (!strncmp(funcfg[i].name, "GPI", 3)) {
```

Comment:
Magic number 3.

Reply:
This is a commit from Aspeed. merged patch using git-apply. future rebase/merge might throw conflict if we edit this.
shall i merge this change as-is?

Reply:
Since, we do plan to get this from Linux upstream in the future. Please fix magic number with macro. As and when Aspeed upstreams to Linux, it is quite possible it will be caught by the review there.

Reply:
Fixed two instances of string comparison